### DISCUSSION TITLE
Joomla Overrides

### DISCUSSION THREAD

**Discussion by brianteeman** [akeeba/panopticon#369] [General]
I am creating an alternative grid view for the main view template and I see that the code for joomla and overrides is in the same file  `ViewTemplates\Main\site_joomla.blade.php` and presented in the same table column

Not a problem for me as I can just put it in its own template file but it seems wrong as template overrides are for both core and extensions and should really be a seperate item imho.

**Comment by nikosdion** [comment]
It does not make sense to put the code for a small badge into a different file. If you do it for one badge you have to do it for all badges. Look at how many different badges there are (even though not all of them will be displayed simultaneously). Also think about the overhead per site per file you have, both from loading a file, and from initialising the site's config (basically, it's parsing the JSON every time you call `$site->getConfig()`).

The way I set it up was such that large conceptual groups have their own file, making them manageable, but smaller and closely related concepts share a file so as not neither overwhelm the maintainer / override developer, nor cause a performance hit due to unnecessary overhead.

As to why overrides are in the code: because this information is retrieved when we are refreshing the _core_ information. Now, think about this for a minute. If you know you should be seeing some overrides in need of attention and they don't show up, what do you do? You'll try to refresh! Which refresh button will you use? The one closest to the badge. So it makes sense to put the badge and the correct refresh button in proximity.

Why not create a separate column? Two reasons. If the table is too wide it doesn't fit a 13" laptop while being readable, _especially when using languages with long language strings, like Greek_. Moreover, Joomla! has overrides, WordPress does not, and it's going to be supported in the future. And before you complain, yes, the "Joomla!" column will become "Core" or "CMS" when the integration takes place. You can already see how newer views are already leaning towards this pattern (and have some code in place for forward-compatibility).

**Reply by brianteeman** [reply]
well as I already said I am creating a grid not a table the columns comment is not applicable here.

**Reply by nikosdion** [reply]
It is very applicable as to why **I did not put this code in a separate file** which is a point that you made.

As I explained, it was **NOT** an arbitrary choice, it was something I thought through.
